toolkit/tools/imagegen/configuration/partitiontabletype.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Parser for the image builder's configuration schemas.
+
+package configuration
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// PartitionTableType is either gpt, mbr, or none
+type PartitionTableType string
+
+const (
+	// PartitionTableTypeGpt selects gpt
+	PartitionTableTypeGpt PartitionTableType = "gpt"
+	// PartitionTableTypeMbr selects mbr
+	PartitionTableTypeMbr PartitionTableType = "mbr"
+	// PartitionTableTypeNone selects no partition type
+	PartitionTableTypeNone PartitionTableType = ""
+)
+
+func (ptt PartitionTableType) String() string {
+	return fmt.Sprintf(string(ptt))
+}
+
+// GetValidPartitionTableTypes returns a list of all the supported
+// disk partition types
+func (ptt *PartitionTableType) GetValidPartitionTableTypes() (types []PartitionTableType) {
+	return []PartitionTableType{
+		PartitionTableTypeGpt,
+		PartitionTableTypeMbr,
+		PartitionTableTypeNone,
+	}
+}
+
+// IsValid returns an error if the PartitionTableType is not valid
+func (ptt *PartitionTableType) IsValid() (err error) {
```

Comment:
I feel like IsValid() shouldn't print errors on bad input, that's its job. Returning an error rather than a bool also allows the caller to print the full context of the issue if they desire.

Wrapping:
```go
errInner := fmt.Errorf("Inner Error")
errOuter := fmt.Errorf("Outer Error, %w, errInner )
// errOuter = "Outer Error, Inner Error"
if errors.Is(errOuter, errInner) {
    //Yes, the outer error "is a" inner error since it includes it.
}
```